lib_gb/src/ppu/gb_ppu.rs

```diff
@@ -286,7 +292,10 @@ impl<GFX:GfxDevice> GbPpu<GFX>{
             }
         }
 
-        let m_cycles_for_state = match self.state{
+        if self.next_state as u8 != self.state as u8{
+            return 1;
+        }
```

Comment:
This currently a hack (together with the next state var), see if you can make more robust